src/til/ut_til/PointTests.cpp

```diff
@@ -221,6 +221,24 @@ class PointTests
         }
     }
 
+    TEST_METHOD(Boolean)
+    {
+        SetVerifyOutput verifyOutputScope{ VerifyOutputSettings::LogOnlyFailures };
+
+        static constexpr til::CoordType values[] = { til::CoordTypeMin, -1, 0, 1, til::CoordTypeMax };
+
+        for (const auto x : values)
+        {
+            for (const auto y : values)
+            {
+                const til::point p{ x, y };
+                const auto expected = x >= 0 && y >= 0;
+                const auto actual = static_cast<bool>(p);

```

Comment:
i'm honestly not sure about the value of this test, since it's just a 1:1 copy of the logic inside `operator bool`.

We had bugs in our VT mouse mode tests _because we copied the mouse VT generation logic_ into the tests, and used that to test the other copy of the logic. :shrug:

Reply:
I personally believe these tests make it more robust because they're the most straight-forward implementation of the check and provide some redundancy with the actual implementation whatever it may be in the future. There's alternative ways to write the emptiness checks which I've explored and these tests helped me make sure that there are no accidental issues on overflow/underflow.